cortado-core/cortado/src/main/java/edu/utexas/cs/utopia/cortado/lockPlacement/LockAssignmentMaxSatEnumerator.java

```diff
@@ -0,0 +1,210 @@
+package edu.utexas.cs.utopia.cortado.lockPlacement;
+
+import com.google.common.collect.Streams;
+import edu.utexas.cs.utopia.cortado.util.sat.enumeration.WeightedPartialMaxSatEnumerator;
+import edu.utexas.cs.utopia.cortado.util.sat.enumeration.WeightedPartialMaxSatEnumeratorFactory;
+import edu.utexas.cs.utopia.cortado.util.sat.formula.PropositionalFormula;
+import edu.utexas.cs.utopia.cortado.util.sat.maxsat.SolveAlreadyAttemptedException;
+import edu.utexas.cs.utopia.cortado.util.sat.maxsat.WeightedPartialMaxSatSolver;
+import edu.utexas.cs.utopia.cortado.util.sat.maxsat.WeightedPartialMaxSatSolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import soot.SootMethod;
+import soot.Unit;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static edu.utexas.cs.utopia.cortado.util.sat.enumeration.WeightedPartialMaxSatEnumerator.EnumerationStatus.NOT_STARTED;
+import static edu.utexas.cs.utopia.cortado.util.sat.enumeration.WeightedPartialMaxSatEnumerator.MultiObjectiveOptimizationPriority.PARETO_FRONT;
+
+/**
+ * Used to enumerate lock assignments along the pareto front
+ */
+public class LockAssignmentMaxSatEnumerator
```

Comment:
Can we remove the enumerator? It hasn't been tested in a long while and  may not be stable